lib/store/cleanup_test.go

```diff
@@ -221,41 +225,110 @@ func TestCleanupManageDiskUsage(t *testing.T) {
 	}
 
-	usage, err := m.scan(op, time.Hour, time.Hour)
-	require.NoError(err)
-	require.Equal(int64(500), usage)
+	s, err = size(op)
+	require.Nil(err)
+	require.Equal(int64(50), s)
 }
 
-func TestCleanupManagerAggressive(t *testing.T) {
+func TestCleanupManagerDiskMetrics(t *testing.T) {
 	require := require.New(t)
 
+	clk := clock.New()
 	config := CleanupConfig{
-		AggressiveThreshold: 80,
-		TTL:                 10 * time.Second,
-		AggressiveTTL:       5 * time.Second,
+		TTI: 1 * time.Hour,
+		TTL: 1 * time.Hour,
 	}
 
-	clk := clock.NewMock()
 	m, err := newCleanupManager(clk, tally.NoopScope)
 	require.NoError(err)
 	defer m.stop()
 
-	_, op, cleanup := fileOpFixture(clk)
+	state, op, cleanup := fileOpFixture(clk)
 	defer cleanup()
 
-	require.Equal(m.checkAggressiveCleanup(op, config, func() (int, error) {
-		return 90, nil
-	}), 5*time.Second)
+	for i := 0; i < 100; i++ {
+		require.NoError(op.CreateFile(core.DigestFixture().Hex(), state, 5))
+	}
+
+	testStats := tally.NewTestScope("", map[string]string{})
+	m.clean(testStats, config, op)
+	snapshot := testStats.Snapshot()
+	usageGauge, ok := snapshot.Gauges()["disk_usage+"]
+	require.True(ok)
+	require.Equal(float64(500), usageGauge.Value())
+}
 
-	require.Equal(m.checkAggressiveCleanup(op, config, func() (int, error) {
-		return 60, nil
-	}), 10*time.Second)
+func TestCleanupManagerCalculateTTL(t *testing.T) {
+	for _, tc := range []struct {
+		desc              string
+		config            CleanupConfig
+		calculateDiskUtil diskUtilFn
+		wantTTL           time.Duration
+	}{
+		{
+			desc: "aggressive cleanup disabled",
+			config: CleanupConfig{
+				TTI: 10 * time.Minute,
+				TTL: 30 * time.Minute,
+			},
+			calculateDiskUtil: nil,
+			wantTTL:           30 * time.Minute,
+		},
+		{
+			desc: "aggressive cleanup enabled, disk util below threshold",
+			config: CleanupConfig{
+				TTI: 10 * time.Minute,
+				TTL: 30 * time.Minute,
+
+				AggressiveThreshold: 50,
+				AggressiveTTL:       5 * time.Minute,
+			},
+			calculateDiskUtil: func(op base.FileOp, c CleanupConfig) (int, error) { return 49, nil },
+			wantTTL:           30 * time.Minute,
+		},
+		{
+			desc: "aggressive cleanup enabled, disk util passed threshold",
+			config: CleanupConfig{
+				TTI: 10 * time.Minute,
+				TTL: 30 * time.Minute,
+
+				AggressiveThreshold: 50,
+				AggressiveTTL:       5 * time.Minute,
+			},
+			calculateDiskUtil: func(op base.FileOp, c CleanupConfig) (int, error) { return 50, nil },
+			wantTTL:           5 * time.Minute,
+		},
+		{
+			desc: "aggressive cleanup enabled, disk util could not be calculated",
+			config: CleanupConfig{
+				TTI: 10 * time.Minute,
+				TTL: 30 * time.Minute,
+
+				AggressiveThreshold: 50,
+				AggressiveTTL:       5 * time.Minute,
+			},
+			calculateDiskUtil: func(op base.FileOp, c CleanupConfig) (int, error) { return 0, errors.New("test-error") },
+			wantTTL:           30 * time.Minute,
+		},
+	} {
+		require := require.New(t)
```

Comment:
we usually do test tables like this:

```
for tt := range tests {
  t.Run(tt.desc, func (t *testing.T) {
    // test body.
  })
}
```

the require := require.New(t) shouldn't be needed